include/stl2/view/iota.hpp

```diff
@@ -68,23 +68,16 @@ STL2_OPEN_NAMESPACE {
 				(Integral<I> && Integral<Bound>) ||
 				SizedSentinel<Bound, I>
 			{
-				return this->bound_ - value_;
+				return bound_ - value_;
 			}
 		};
 
-		template <WeaklyIncrementable I>
-		explicit iota_view(I value) -> iota_view<I>;
-
-		template <Incrementable I>
-		iota_view(I value, I bound) -> iota_view<I, I>;
-
-		template <WeaklyIncrementable I, Semiregular Bound>
-		requires WeaklyEqualityComparable<I, Bound> && !ConvertibleTo<Bound, I>
-		iota_view(I value, Bound bound) -> iota_view<I, Bound>;
+		template <Incrementable I, ConvertibleTo<I> Bound>
+		iota_view(I value, Bound) -> iota_view<I, I>;
```

Comment:
No, this is _not_ semantically equivalent, and it's not what I want. I want `iota_view{0, v.size()}` to be an error. Users should have to disambiguate with a cast: `iota_view{0, (int)v.size()}`, or some such.

Reply:
Yes, we discussed that in the other PR. I'll remove this guide.

Reply:
> I'll remove this guide.

...and kill the implicit guide for the `iota_view(I, Bound)` constructor, and restore the original set of explicit guides, so that you get your ambiguity.